librocketchat.c

```diff
@@ -2203,7 +2271,13 @@ rc_close(PurpleConnection *pc)
 	g_return_if_fail(ya != NULL);
 	
 	// account = purple_connection_get_account(pc);
-	if (ya->websocket != NULL) purple_ssl_close(ya->websocket);
+	if (ya->websocket != NULL)
+		purple_ssl_close(ya->websocket);
+	else if (ya->fd > 0) {
+		if (ya->pc->inpa)

```

Comment:
is `pc->inpa` exposed through a function rather than direct struct access?  If so can we use that func for forwards compatibility?

Reply:
Purples Jabber plugin also uses the same way to access pc. The pc is part of the account structure, I'm not sure how that would change with Purple 3.x but I don't see any of that so far.

Reply:
Purple 3 makes all the eg `PurpleConnection` and `PurpleAccount` structs private

Reply:
Oh it seams I should have used `ya->inpa` instead.
Purple 3.x would do `g_source_remove(ya->inpa).`